### PR TITLE
A Raft of Fixes To Private Dependencies

### PR DESCRIPTION
**Pull request by CodaFi**
- Turn the dependency verifier loose on the entire compiler pipeline, not just the typechecker
- Fixup the getter for the active dependency source. It was mistakenly set to read the source stack the old way when I refactored this in #31960 
- Make LookupInModuleRequest a dependency sink
- Commit a tricky regression test that verifies we always take a dependency on the underlying type of a typealias.
- Perform the unioning step during request replay.

Resolves rdar://64008262

### CODE REVIEW

**Comment by CodaFi** [comment]
@swift-ci smoke test

File: lib/Sema/TypeCheckDeclPrimary.cpp [hunk 1]
```diff
@@ -1692,8 +1692,7 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
 
   void visitTypeAliasDecl(TypeAliasDecl *TAD) {
     // Force requests that can emit diagnostics.
-    (void) TAD->getGenericSignature();
-    (void) TAD->getUnderlyingType();
+    (void) TAD->getInterfaceType();
```
Comment (slavapestov) [inline lib/Sema/TypeCheckDeclPrimary.cpp:1695]: This change should not be necessary

**Review by slavapestov — APPROVED** [review]

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Comment by CodaFi** [comment]
@swift-ci test Windows

**Comment by CodaFi** [comment]
Just to see what happens here

@swift-ci test compiler performance

**Comment by CodaFi** [comment]
@swift-ci test Windows

**Comment by CodaFi** [comment]
@swift-ci test compiler performance

**Comment by CodaFi** [comment]
@swift-ci test Windows

**Comment by CodaFi** [comment]
(Oops, let's cancel the perf test. I meant to push a different stack of commits before that).

**Comment by CodaFi** [comment]
⛵

**Pull request merged**